client/src/components/Markdown/MarkdownDialog.vue

```diff
@@ -1,3 +1,223 @@
+<script setup lang="ts">
+import axios from "axios";
+import BootstrapVue from "bootstrap-vue";
+import Vue, { computed, ref } from "vue";
+
+import { getAppRoot } from "@/onload/loadConfig";
+import { getCurrentGalaxyHistory } from "@/utils/data";
+
+import MarkdownSelector from "./MarkdownSelector.vue";
+import MarkdownVisualization from "./MarkdownVisualization.vue";
+import DataDialog from "@/components/DataDialog/DataDialog.vue";
+import BasicSelectionDialog from "@/components/SelectionDialog/BasicSelectionDialog.vue";
+import DatasetCollectionDialog from "@/components/SelectionDialog/DatasetCollectionDialog.vue";
+
+Vue.use(BootstrapVue);
+
+interface MarkdownDialogProps {
+    argumentName?: string;
+    argumentType?: string;
+    argumentPayload?: object;
+    labels: string[];
+    useLabels: boolean;
+}
+
+const props = withDefaults(defineProps<MarkdownDialogProps>(), {
+    argumentName: null,
+    argumentType: null,
+    argumentPayload: null,
```

Comment:
```suggestion
    argumentName: undefined,
    argumentType: undefined,
    argumentPayload: undefined,
```
Or make the props nullable with `| null` ;)

Reply:
Why don't I ever understand the difference between these. It yells at me until I find a stable state and the stable state is definitely always wrong 😆.